.github/workflows/validate-lsp.yml

```diff
@@ -0,0 +1,44 @@
+name: Validate that changes doesn't break elm-language-server
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        node-version: [18, 20]
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          path: "tree-sitter-elm"
+      - uses: actions/checkout@v4
+        name: Checkout main branch of elm-language-server
+        with:
+          repository: "elm-tooling/elm-language-server"
+          path: "elm-language-server"
+      - name: Install tree-sitter dependencies and generate wasm bundle
+        run: |
+          cd tree-sitter-elm/
+          npm i
+          npm run build
+          npx tree-sitter build-wasm
+          mv ./tree-sitter-elm.wasm ../elm-language-server/tree-sitter-elm.wasm -f
+
+      - name: Install elm-language-server dependencies, compile, and run tests
+        run: |
+          cd elm-language-server/
+          npm i
+          npm run compile
+          npm install -g elm-format
```

Comment:
@razzeee I saw that this was done in the [elm-language-server test pipeline](https://github.com/elm-tooling/elm-language-server/blob/9dbe00839720908d471ccdd9f4dd6e5ab89cfead/.github/workflows/test.yaml#L31) as well, but I'm just wondering if it wouldn't be better to add elm-format to the devDependencies for elm-language-server.

Currently if you checkout the repo and try to run the test suite without having elm-format installed globally, the test in test/diff.test.ts in elm-language-server fails.

Reply:
I'm actually unsure, has been some time since we looked at this and the binary is not exactly the same thing as the node dependency if I remember correctly, it has different performance etc.

It might work fine, if it's only for the tests though.